File: frontend/gearmind_app/lib/models/product.dart
class Product {
  const Product({
    required this.id,
    required this.name,
    required this.brand,
    required this.category,
    required this.price,
    required this.rating,
    required this.sales,
    required this.imageUrl,
    required this.tags,
    required this.description,
    required this.specs,
    required this.pros,
    required this.cons,
    required this.recommendedFor,
    required this.notRecommendedFor,
    required this.reviewSummary,
    required this.matchScore,
  });

  final String id;
  final String name;
  final String brand;
  final String category;
  final double price;
  final double rating;
  final int sales;
  final String imageUrl;
  final List<String> tags;
  final String description;
  final Map<String, dynamic> specs;
  final List<String> pros;
  final List<String> cons;
  final List<String> recommendedFor;
  final List<String> notRecommendedFor;
  final String reviewSummary;
  final int matchScore;

  factory Product.fromJson(Map<String, dynamic> json) {
    List<String> strings(String key) =>
        (json[key] as List<dynamic>? ?? const []).map((e) => '$e').toList();
    return Product(
      id: json['id'] as String,
      name: json['name'] as String,
      brand: json['brand'] as String,
      category: json['category'] as String,
      price: (json['price'] as num).toDouble(),
      rating: (json['rating'] as num).toDouble(),
      sales: (json['sales'] as num).toInt(),
      imageUrl: json['image_url'] as String? ?? json['imageUrl'] as String,
      tags: strings('tags'),
      description: json['description'] as String,
      specs: Map<String, dynamic>.from(json['specs'] as Map? ?? const {}),
      pros: strings('pros'),
      cons: strings('cons'),
      recommendedFor: strings('recommended_for'),
      notRecommendedFor: strings('not_recommended_for'),
      reviewSummary: json['review_summary'] as String,
      matchScore: (json['match_score'] as num).toInt(),
    );
  }
}

File: app/models/product.py
"""Product persistence model."""

from __future__ import annotations

from datetime import datetime
from decimal import Decimal
from typing import Any, Optional

from sqlalchemy import BigInteger, DateTime, Index, Integer, JSON, Numeric, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class Product(Base):
    __tablename__ = "products"
    __table_args__ = (
        Index("ix_products_category", "category"),
        Index("ix_products_price", "price"),
    )

    id: Mapped[int] = mapped_column(
        BigInteger().with_variant(Integer, "sqlite"),
        primary_key=True,
        autoincrement=True,
    )
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    category: Mapped[Optional[str]] = mapped_column(String(64))
    brand: Mapped[Optional[str]] = mapped_column(String(64))
    sku: Mapped[Optional[str]] = mapped_column(String(128))
    price: Mapped[Optional[Decimal]] = mapped_column(Numeric(10, 2))
    original_price: Mapped[Optional[Decimal]] = mapped_column(Numeric(10, 2))
    platform: Mapped[Optional[str]] = mapped_column(String(64))
    product_url: Mapped[Optional[str]] = mapped_column(String(1024))
    image_url: Mapped[Optional[str]] = mapped_column(String(512))
    image_urls: Mapped[Optional[Any]] = mapped_column(JSON)
    rating: Mapped[Optional[Decimal]] = mapped_column(Numeric(3, 2))
    sales: Mapped[Optional[int]] = mapped_column(Integer)
    description: Mapped[Optional[str]] = mapped_column(Text)
    specs: Mapped[Optional[Any]] = mapped_column(JSON)
    tags: Mapped[Optional[Any]] = mapped_column(JSON)
    suitable_scene: Mapped[Optional[Any]] = mapped_column(JSON)
    stock: Mapped[Optional[int]] = mapped_column(Integer)
    stock_status: Mapped[Optional[str]] = mapped_column(String(32))
    review_summary: Mapped[Optional[str]] = mapped_column(Text)
    created_at: Mapped[Optional[datetime]] = mapped_column(DateTime)
